src/core/tensor/tensor.cc

```diff
@@ -94,22 +128,36 @@ void Tensor::ResetLike(const Tensor &in) {
   shape_ = in.shape_;
 }
 
+//yisen todo
+//if tensor is not transposed yet i.e strides == 1, then we simply change the shape and generate new default strides
+//if tensor is already transposed i.e strides != 1, it should be copied to a new tensor with newly generated default strides 
 void Tensor::Reshape(const Shape &shape) {
   if (Product(shape_) != Product(shape)) {
```

Comment:
just add CHECK_EQ(Product(shape_), Product(shape)) << "Reshape requires the original shape and new shape have the same volume." << vector2str(shape_) << vector2str(shape);
implement 
`string vector2str(const vector<int>& shape);`